kani-driver/src/exe_trace.rs

```diff
@@ -0,0 +1,264 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use crate::session::KaniSession;
+use kani_metadata::HarnessMetadata;
+use std::collections::hash_map::DefaultHasher;
+use std::ffi::OsString;
+use std::fs;
+use std::hash::{Hash, Hasher};
+use std::path::Path;
+use std::process::Command;
+
+/// Module for parsing deterministic values from CBMC output traces,
+/// generating executable traces, and adding them to the user's source code.
+
+impl KaniSession {
+    /// The main driver for generating executable traces and adding them to source code.
+    pub fn exe_trace_main(&self, specialized_obj: &Path, harness: &HarnessMetadata) {
+        if self.args.gen_exe_trace {
+            let (det_vals, interp_det_vals) = parser::get_det_vals(specialized_obj);
+            let (exe_trace, exe_trace_func_name) =
+                format_unit_test(&harness.mangled_name, &det_vals[..], &interp_det_vals[..]);
+
+            println!("Executable trace:\n");
+            println!("{}", exe_trace);
+
+            if self.args.add_exe_trace_to_src {
+                if !self.args.quiet {
+                    println!("Now modifying the source code to include the executable trace.");
+                }
+                let proof_harness_end_line: usize = harness
+                    .original_end_line
+                    .parse()
+                    .expect("Couldn't convert proof harness line from str to usize");
```

Comment:
Maybe change the error to "Invalid proof harness end line: {line}"

Reply:
Done